covscan: memory leak in audit log when adding entries

**Issue Description**
Covscan reports:
```
1. Defect type: RESOURCE_LEAK
19. 389-ds-base-2.2.4/ldap/servers/slapd/auditlog.c:231: leaked_storage: Variable "vals" going out of scope leaks the storage it points to.
```
And code review confirms that a charray_free is missing. 
Impact: if nsslapd-auditlog-display-attrs is not *  all added entries attributes values that are logged in audit log are leaking. 

**Package Version and Platform:**
 - Platform: RedHat
 - Package and version: 389-ds-base-2.2.4

**Steps to Reproduce**
Steps to reproduce the behavior:
1. Generate source package 
2. Have a covscan run using these sources
3. Look at the report about the"Resources leaks" section

**Expected results**
No leaks should be reported.

@progier389 - looks like this introduced a compiler warning:

```
../389-ds-base/ldap/servers/slapd/auditlog.c: In function 'add_entry_attrs':
../389-ds-base/ldap/servers/slapd/auditlog.c:257:27: warning: unused variable 'vals' [-Wunused-variable]
  257 |             Slapi_Value **vals = attr_get_present_values(entry_attr);
      |                           ^~~~
```

--- a/ldap/servers/slapd/auditlog.c
+++ b/ldap/servers/slapd/auditlog.c
@@ -250,7 +250,6 @@ add_entry_attrs(Slapi_Entry *entry, lenstr *l)
             if (entry_attr) {
                 log_entry_attr(entry_attr, req_attr, l);
             }
-            charray_free(vals);
         }
     } else {
         /* Return all attributes */

--- a/ldap/servers/slapd/auditlog.c
+++ b/ldap/servers/slapd/auditlog.c
@@ -250,6 +250,7 @@ add_entry_attrs(Slapi_Entry *entry, lenstr *l)
             if (entry_attr) {
                 log_entry_attr(entry_attr, req_attr, l);
             }
+            charray_free(vals);
         }
     } else {
         /* Return all attributes */

--- a/ldap/servers/slapd/auditlog.c
+++ b/ldap/servers/slapd/auditlog.c
@@ -254,7 +254,6 @@ add_entry_attrs(Slapi_Entry *entry, lenstr *l)
     } else {
         /* Return all attributes */
         for (; entry_attr; entry_attr = entry_attr->a_next) {
-            Slapi_Value **vals = attr_get_present_values(entry_attr);
             char *attr = NULL;
 
             slapi_attr_get_type(entry_attr, &attr);

